File: backendSpringboot/k-medical/src/main/java/com/medical/model/Invoice.java
package com.medical.model;

import java.util.Date;
import java.util.List;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Invoice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Date dateIssued;
    private double totalAmount;
    private double prixConsultation;  
    @OneToOne
    @JoinColumn(name = "consultation_id")
    private Consultations consultation;
 
}

File: frontendAngular/medical/k-medical/src/app/monClass/Invoice.ts
import { Consultations } from "./Consultations";
import { Treatment } from "./Treatment";

export class Invoice {
  id: number;
  dateIssued: string;
  totalAmount: number;
  consultation: Consultations; // Define a more specific type if available
  treatments: Treatment[];
  prixConsultation:number
  }